ssvsigner/client.go

```diff
@@ -23,30 +25,41 @@ type Client struct {
 	httpClient *http.Client
 }
 
-func NewClient(baseURL string, opts ...ClientOption) *Client {
-	baseURL = strings.TrimRight(baseURL, "/")
+// ClientOption is used to handle client options.
+type ClientOption func(*Client) error
 
-	c := &Client{
-		baseURL: baseURL,
-		httpClient: &http.Client{
-			Timeout: 30 * time.Second,
-		},
-		logger: zap.NewNop(),
-	}
+// WithLogger sets a custom logger for the client.
+func WithLogger(logger *zap.Logger) ClientOption {
+	return func(client *Client) error {
+		client.logger = logger
 
-	for _, opt := range opts {
-		opt(c)
+		return nil
 	}
+}
 
-	return c
+// WithTLS is for setting the TLS.
+func WithTLS(certPath, keyPath, caPath string) ClientOption {
+	return func(client *Client) error {
+		return client.configureTLS(certPath, keyPath, caPath)
+	}
 }
```

Comment:
imo we should ask for the types of the decoded keys rather than file paths at this point

the parsing can happen in the main package where all the others file paths are loaded